src/server/saved_objects/saved_objects_mixin.js

```diff
@@ -26,11 +26,30 @@ export function savedObjectsMixin(kbnServer, server) {
   server.route(createGetRoute(prereqs));
   server.route(createUpdateRoute(prereqs));
 
+  async function onBeforeWrite() {
+    const adminCluster = server.plugins.elasticsearch.getCluster('admin');
+
+    try {
+      await adminCluster.callWithInternalUser('cluster.health', {
+        timeout: `${server.config().get('savedObjects.indexCheckTimeout')}ms`,
+        index: server.config().get('kibana.index'),
+        waitForStatus: 'yellow',
+      });
+    } catch (error) {
+      if (error && error.body && error.body.status === 'red') {
+        server.log(['debug', 'savedObjects'], `Attempted to write to the Kibana index when it didn't exist.`);
+        throw new adminCluster.errors.NotFound();
```

Comment:
I find this error misleading - given the index is unavailable, we will fail API create requests with a 404, not found.

Maybe a 503 service unavailable with a message specific to the Kibana index being unavailable?

Reply:
I totally agree, 404 does not make sense for create requests. I'll update the create method to send a 503 instead of a 404. 

> note to the reader: We chatted about this over video and worked out that all of the other APIs can still respond with 404 when the "target data is does not exist", regardless if that is because of the index existing or not.